### PR TITLE
Persons index improvements

### LINKED ISSUE #735: Small bug in Person search
From Ron:

> Go to Persons page.
> Type Ron and click Search
> Scroll to second result page
> Type Feliks and click Search
> No result found.
> 
> I think the script still thinks I am on second page.

### PR DESCRIPTION
**Pull request by jonatanklosko**
Fixes #735 and adds pagination information in the url.

### CODE REVIEW

**Comment by jfly** [comment]
Nit saying you should do this as part of this PR, but could we add the pagination information to the url? That way people can copy paste urls.

**Comment by jonatanklosko** [comment]
@jfly great idea! Done =)

File: WcaOnRails/app/assets/javascripts/persons.js [hunk 1]
```diff
@@ -31,5 +33,12 @@ onPage('persons#index', function() {
 
   $table.on('load-success.bs.table', function(e, data) {
     $('#search-box i').removeClass('fa-spinner fa-spin').addClass('fa-search');
+
+    // Update params in the url.
+    var params = personsTableAjax.queryParams(); // Get region and search params.
+    params.page = $table.bootstrapTable('getOptions').pageNumber;
+    var url = location.toString();
+    url = url.replace(/persons.*/, 'persons?' + $.param(params));
```
Comment (jfly) [inline WcaOnRails/app/assets/javascripts/persons.js]: I'm not sure I fully understand this line. Does this assume that persons is the first url parameter?

Comment (jonatanklosko) [inline WcaOnRails/app/assets/javascripts/persons.js]: 'persons' is the last *constant* part of the path: 'https://worldcubeassociation.org/persons', so we grab it and all params after it and replace this with the new part.

Comment (jfly) [inline WcaOnRails/app/assets/javascripts/persons.js]: I wish we could use a javascript library to explicitly change just the url parameters, but this seems fine for now.

**Comment by jfly** [comment]
LGTM!

**Pull request merged**